tests/components/advantage_air/test_switch.py

```diff
@@ -0,0 +1,79 @@
+"""Test the Advantage Air Switch Platform."""
+
+from json import loads
+
+from homeassistant.components.advantage_air.const import (
+    ADVANTAGE_AIR_STATE_OFF,
+    ADVANTAGE_AIR_STATE_ON,
+)
+from homeassistant.components.switch import (
+    DOMAIN as SWITCH_DOMAIN,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.const import ATTR_ENTITY_ID, STATE_OFF
+
+from tests.components.advantage_air import (
+    TEST_SET_RESPONSE,
+    TEST_SET_URL,
+    TEST_SYSTEM_DATA,
+    TEST_SYSTEM_URL,
+    add_mock_config,
+)
+
+
+async def test_cover_async_setup_entry(hass, aioclient_mock):
+    """Test climate setup without sensors."""
+
+    aioclient_mock.get(
+        TEST_SYSTEM_URL,
+        text=TEST_SYSTEM_DATA,
+    )
+    aioclient_mock.get(
+        TEST_SET_URL,
+        text=TEST_SET_RESPONSE,
+    )
+
+    await add_mock_config(hass)
+
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    assert len(aioclient_mock.mock_calls) == 1
+
+    # Test Switch Entity
+    entity_id = "switch.ac_one_fresh_air"
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == STATE_OFF
+
+    entry = registry.async_get(entity_id)
+    assert entry
+    assert entry.unique_id == "uniqueid-ac1-freshair"
+
+    await hass.services.async_call(
+        SWITCH_DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: [entity_id]},
+        blocking=True,
+    )
+    assert len(aioclient_mock.mock_calls) == 3
+    assert aioclient_mock.mock_calls[-2][0] == "GET"
+    assert aioclient_mock.mock_calls[-2][1].path == "/setAircon"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)